data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqs/RawSqsMessageHandler.java

```diff
@@ -0,0 +1,76 @@
+package org.opensearch.dataprepper.plugins.source.sqs;
+
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.opensearch.dataprepper.buffer.common.BufferAccumulator;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.sqs.model.Message;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.time.Instant;
+import java.util.Objects;
+
+/**
+ * Implements the SqsMessageHandler to read and parse SQS messages generically and push to buffer.
+ */
+public class RawSqsMessageHandler implements SqsMessageHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RawSqsMessageHandler.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * Processes the SQS message, attempting to parse it as JSON, and adds it to the buffer.
+     *
+     * @param message            - the SQS message for processing
+     * @param url                - the SQS queue url
+     * @param bufferAccumulator  - the buffer accumulator
+     * @param acknowledgementSet - the acknowledgement set for end-to-end acknowledgements
+     */
+    @Override
+    public void handleMessage(final Message message,
+                              final String url, 
+                              final BufferAccumulator<Record<Event>> bufferAccumulator,
+                              final AcknowledgementSet acknowledgementSet) {
+        try {
+            ObjectNode dataNode = objectMapper.createObjectNode();
+            dataNode.set("message", parseMessageBody(message.body()));
+            dataNode.put("queueUrl", url);
+
+            Instant now = Instant.now();
+            int unixTimestamp = (int) now.getEpochSecond(); 
+            dataNode.put("sentTimestamp", unixTimestamp);
```

Comment:
Let's add this as metadata instead of as a event property. See how the Kafka source does this.

Also, you are getting the received time here. The sentTimestamp comes from the SQS message itself as an attribute.